File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/GUI/FabricaMAC.php
<?php

namespace GUI;

use Componentes\Boton;
use Componentes\Checkbox;
use Componentes\MAC\BotonMAC;
use Componentes\MAC\CheckboxMAC;

class FabricaMAC implements FabricaGUI {
	function crearBoton(): Boton {
		return new BotonMAC;
	}

	function crearCheckbox(): Checkbox {
		return new CheckboxMAC;
	}
}

File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/GUI/FabricaMAC.java
import Componentes.Botones.Boton;
import Componentes.Botones.BotonMAC;
import Componentes.Checkboxes.Checkbox;
import Componentes.Checkboxes.CheckboxMAC;

class FabricaMAC implements FabricaGUI {
	public Boton crearBoton() {
		return new BotonMAC();
	}

	public Checkbox crearCheckbox() {
		return new CheckboxMAC();
	}
}
